### ISSUE TITLE
iOS - Memory goes up until it crashes

### ISSUE THREAD

**Issue by pepperav** [ismaelsousa/vision-camera-ocr#17]
Hello @ismaelsousa,
I'm struggling with memory leaks.

It happens when I start to invoke ML plugin inside the frameProcessor flow. 

_When I comment that line, the memory is stable._

See below 👇

```
const onScanResponse = Worklets.createRunOnJS((frame: Frame, scannedOcr: OCRFrame) => {
        // Do something
    });

  const frameProcessor = useSkiaFrameProcessor(frame => {
      'worklet'

      frame.render();

      runAsync(frame, () => {
          'worklet'

          try {
              // 👇👇👇 Uncomment here to get memory leaks and crash!!!!!
              // const scannedOcr = scanOCR(frame);
              
              // do something...
              
          } catch (error) {
              console.log("❌", error);
          }
      });
    }, [ onScanResponse ]);
```

**package.json**:
```
{
  "@ismaelmoreiraa/vision-camera-ocr": "^3.0.1",
  "react-native": "0.74.3",
  "react-native-vision-camera": "^4.5.0",
  "react-native-worklets-core": "^1.3.3"
}
```

https://github.com/user-attachments/assets/6f1c251b-9ece-46cd-a0c9-a8e058e02b4b

**Comment by ismaelsousa** [comment]
Cool, thanks I will check this :)

**Comment by pepperav** [comment]
Ehi @ismaelsousa, any update? 

Is there anything I can do to help? 🙂

**Comment by pepperav** [comment]
Ehi @ismaelsousa ,
I found a possible solution using only `VisionImage` by `frame.buffer`

```
public override func callback(_ frame: Frame, withArguments arguments: [AnyHashable: Any]?) -> Any? {
        
        let visionImage = VisionImage(buffer: frame.buffer)
        
        // Set the orientation of visionImage to the opposite of the frame's orientation
        // Opposite compare to the `up` orientation
        switch frame.orientation {
            case .left:
                visionImage.orientation = .right
            case .right:
                visionImage.orientation = .left
            case .up, .down:
                fallthrough
            
            default: visionImage.orientation = frame.orientation
        }
        
        var result: Text
        
        do {
          result = try OCRFrameProcessorPlugin.textRecognizer.results(in: visionImage)
        } catch let error {
          print("Failed to recognize text with error: \(error.localizedDescription).")
          return nil
        }
        
        return [
            "result": [
                "text": result.text,
                "blocks": OCRFrameProcessorPlugin.getBlockArray(result.blocks),
            ]
        ]
    }
```

I would like to do a pr but I get this error:
```
remote: Permission to ismaelsousa/vision-camera-ocr.git denied to pepperav.
fatal: unable to access 'https://github.com/ismaelsousa/vision-camera-ocr.git/': The requested URL returned error: 403
```

**Comment by ismaelsousa** [comment]
@pepperav did you make a fork?

**Comment by pepperav** [comment]
nop! I will retry with a fork

**Comment by ismaelsousa** [comment]
Great 😊

**Comment by pepperav** [comment]
Done! @ismaelsousa  pls check

thank you

**Comment by ismaelsousa** [comment]
Available in the pre-release: 3.0.2-1
